avocado/core/test.py

```diff
@@ -171,11 +171,11 @@ def record_and_warn(*args, **kwargs):
         if name is not None:
             if not isinstance(name, TestName):
                 raise NameNotTestNameError(name)
-            self.name = name
+            self.__name = name
```

Comment:
OK, with the double underscore indeed people will have to try harder to touch those variables.